cheroot/server.py

```diff
@@ -789,6 +789,11 @@ def read_request_line(self):
                 return False
             rp = req_protocol[5:].split(b'.', 1)
             rp = tuple(map(int, rp))  # Minor.Major must be threat as integers
+            if len(rp) != 2:
+                self.simple_response(
+                    '400 Bad Request', 'Malformed Request-Line: bad version',
```

Comment:
I'm a bit indecisive between 400 and 505 here. Do you have an RFC reference confirming the correct behavior?

Reply:
Alright, I've checked what a few major web-sites return (twitter/github/google/youtube/etc.) and they all use 400.